packages/cadl-ranch-specs/http/azure/core/basic/main.tsp

```diff
@@ -214,3 +238,16 @@ Expected response body:
 ```
 """)
 op export is ResourceOperations.ResourceAction<User, UserExportParams, User>;
+
+/** Swaps two existing dogs with each other. */
+@scenario
+@scenarioDoc("""
+Should only generate models named User and UserOrder.
+Expected query parameter: api-version=2022-12-01-preview
+""")
+@collectionAction(Dog, "swapdogs")
+op swapdogs is ResourceOperations.ResourceCollectionAction<
```

Comment:
swapdogs...

Could we use test with more meaningful API?

Reply:
fixed